app/src/ui/history/selected-commit.tsx

```diff
@@ -383,20 +385,21 @@ export class SelectedCommit extends React.Component<
 
     items.push({
       label: viewOnGitHubLabel,
-      action: () => this.onViewOnGitHub(file),
+      action: () => this.onViewOnGitHub(selectedCommits[0].sha, file),
       enabled:
-        !this.props.isLocal &&
+        // TODO: maybe we can assume to use last commit in multi commit
+        // scenario?
+        selectedCommits.length === 1 &&
```

Comment:
Thoughts? I believe this should be true. Commits A, B, C and a diff between them (assuming C is the latest), if you look at the file as it is in C, it should be as it is displayed in the diff. The only thing is that we would have to make the localCommitShas verify all the commits in the selection are not local to know it is the same as reflected on dotcom.

Reply:
Nvm... I tried it.. since we are specifying the commit in the url.. then we can only reference files in that commit. Thus, if commit A has file X and commit B has file Y. Then, you can't to view file on GitHub for X given commit B that has no file X.

Reply:
We could throw them to the compare view.. but doesn't appear we can just single out a file that way... so I think it makes sense to just leave this disabled for a multi commit.. or remove altogether for multi commit.

Reply:
Refactor SelectedCommit to be SelectedCommits for multi commit diffing